src/store/index.js

```diff
@@ -14,6 +14,33 @@ export function get (url) {
   return fetch(url, options)
 }
 
+// this function always returns an object. in case of success, the object is
+// the JSON sent by the server. in case of any errors, the `error` property
+// contains a description of the error.
```

Comment:
introducing a new pattern here, might be nice to update other server-interaction functions to also work like this.

Reply:
Ooph.  I don't want to bog this PR down in a semantics debate, but I prefer the existing pattern.

Being able to use promise rejection to differentiate different failure modes is convenient.  That said, we aren't really investing heavily in error-handling at present!

i.e. You can do cool stuff like this

```
getJson('/api/endpoint')
  .catch(transportFailure => { return retry(); })
  .then(response => {
    if (!response.ok) {
      return Promise.reject();
    }
    // handle success
  })
  .catch(serverError => {
    // handle server error
  })
```

Reply:
mmm, i guess i don't really like the promise syntax, i find promises to be hard to think about, and really prefer async/await syntax. guaranteeing that this function will always return an object helps simplify the call sites, since generally we just log the error to an alert modal. but i also like the example you have here, seems pretty concise. i'm not totally sure how to implement that in the `getJson` function.

Reply:
i guess in python, i would declare a few exception classes, and throw relevant exceptions; i still find that easier to read than chained promise syntax, but maybe it's not very java-scripty? maybe mostly because you just always have one `catch` after `try` in js, and i guess you have to check `instanceof` inside the `catch` vs catching specific exceptions by type

Reply:
I think you can do something similar with async/await syntax.  The end goal is just to have a good mechanism to distinguish transport and server exception types.

And yeah, you're right. In python I'd definitely use exception subclasses!  I guess transport = reject / server = not ok just mimics the fetch API, so that's what I expect when I'm poking around a new codebase.

Again, we're not really handling transport failures anywhere else at all at the moment, so I don't feel that strongly.  😅